crates/storage_models/src/payment_intent.rs

```diff
@@ -231,20 +232,25 @@ impl From<PaymentIntentUpdate> for PaymentIntentUpdateInternal {
                 amount_captured,
                 // customer_id,
                 return_url,
-                client_secret: make_client_secret_null_if_success(Some(status)),
+                client_secret: make_client_secret_null_based_on_status(Some(status)),
                 modified_at: Some(common_utils::date_time::now()),
                 ..Default::default()
             },
         }
     }
 }
 
-fn make_client_secret_null_if_success(
+fn make_client_secret_null_based_on_status(
     status: Option<storage_enums::IntentStatus>,
 ) -> Option<Option<String>> {
-    if status == Some(storage_enums::IntentStatus::Succeeded) {
-        Some(None)
-    } else {
-        None
-    }
+    status.and_then(|status| match status {
+        storage_enums::IntentStatus::Succeeded
+        | storage_enums::IntentStatus::Failed
+        | storage_enums::IntentStatus::Cancelled => Some(None),
+        storage_enums::IntentStatus::Processing
+        | storage_enums::IntentStatus::RequiresCustomerAction
+        | storage_enums::IntentStatus::RequiresPaymentMethod
+        | storage_enums::IntentStatus::RequiresConfirmation
+        | storage_enums::IntentStatus::RequiresCapture => None,
```

Comment:
could have used `_` pattern here

Reply:
Didn't use `_` pattern because if a new enum variant is added in future they won't be forced by compiler to handle the case here.